biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -115,13 +242,13 @@ public void setProperties(Map<String,String> map) throws Exception {
 
 		if (configuration.pom() != null) {
 			List<String> parts = Strings.split(configuration.pom());
-			pomFiles = new ArrayList<>();
+			pomFiles = new HashMap<>();
```

Comment:
Use a LinkedHashMap to preserve iteration order. This is important.

Reply:
Out of curiosity: why is the iteration order important for Bnd?

Reply:
For consistent across builds across different systems. You are replacing a List, which is ordered, by a HashMap whose keySet is this unordered. This is a source of entropy we need to avoid. Thus you will see that Bnd almost always uses LinkedHashMap in preference to just HashMap to provide a consistent iteration order.